127 fix makerspace components styling

katiekhamarkhanova opened the pull request:
made it so that articles just display image, headline, and short portion of text (if too much text, it gets cut off by ...)
when user clicks on article, if the article has an external link they are brought to that but if no external link, a popup appears with the full article text (scroll to read) along with the headline and image 
+ prettier code formatting

katiekhamarkhanova committed:
wip

katiekhamarkhanova committed:
a

katiekhamarkhanova committed:
click to read more on articles + prettier formatting

jo-svg commented:
<img width="888" height="90" alt="Screenshot 2025-12-17 at 12 16 45 PM" src="https://github.com/user-attachments/assets/844697e2-afb1-42e8-b7b1-1f8453346768" />
is this supposed to be empty lmfao

katiekhamarkhanova commented:
no it is not idk why it is like that

katiekhamarkhanova commented:
think i didn't push when i worked from home

katiekhamarkhanova commented:
😢

joe-238 commented:
it still empty did you push it yet lol

katiekhamarkhanova commented:
oh wait nah my brain ain't working i'm so tired

it is supposed to be empty i think i wanted to have the full makerspace article popup as a separate component but i just ended up putting it in the makerspaceGrid.vue file

katiekhamarkhanova committed:
delete makerspacePopup.vue

katiekhamarkhanova commented:
deleted the file

jo-svg commented:
@katiekhamarkhanova REMINDER TO YOU AND ME THAT I ASKED YOU TO MAKE IT A COMPONENT

katiekhamarkhanova committed:
components

put the pagination and full article popup into separate components for organizzation

jo-svg commented on siths-redesign/components/FullArticle.vue:
```diff
@@ -0,0 +1,42 @@
+        <h2 class="text-gray-900 pr-4 text-2xl font-bold">{{ article.headline }}</h2>
+        <button
+          @click="$emit('close')"
+          class="text-gray-400 hover:text-gray-600 p-1 transition-colors hover:cursor-pointer"
```
Theres no icon or context to this button

jo-svg commented on siths-redesign/components/FullArticle.vue:
```diff
@@ -0,0 +1,42 @@
+defineProps({
+  article: {
+    type: Object,
+    default: null
```
Maybe change this ot required: true instead of default null

jo-svg commented on siths-redesign/components/FullArticle.vue:
```diff
@@ -0,0 +1,42 @@
+        <img
+          v-if="article.imageUrl"
+          :src="article.imageUrl"
+          :alt="article.headline"
```
Could be more descriptive

jo-svg commented on siths-redesign/components/FullArticle.vue:
```diff
@@ -0,0 +1,42 @@
+      @click.stop
+    >
+      <div class="border-gray-200 flex items-start justify-between border-b p-6">
+        <h2 class="text-gray-900 pr-4 text-2xl font-bold">{{ article.headline }}</h2>
```
Nitpick idk if prettier did this but i think doing pr-4 text-2xl font-bold text-gray-900 is better

kennething replied:
bros going after the CLASS ORDER nah thats crazy even i dont nitpick that hard :shipit:

jo-svg commented on siths-redesign/components/MakerspaceGrid.vue:
```diff
@@ -23,78 +19,83 @@
 </template>
 
 <script setup>
+import FullArticle from './FullArticle.vue'
```
Imports while using nuxt?

jo-svg commented on siths-redesign/components/MakerspaceGrid.vue:
```diff
@@ -1,20 +1,16 @@
+          class="flex flex-col overflow-hidden rounded-lg bg-white text-black shadow-lg transition-all duration-300 hover:shadow-2xl md:flex-row"
         >
           <div
             v-if="article.imageUrl"
```
Feels like unnecessary div, you xan js add this to img

jo-svg commented on siths-redesign/components/MakerspaceGrid.vue:
```diff
@@ -1,20 +1,16 @@
-      <div
-        v-for="article in paginatedArticles"
+      <a
+        v-for="article in paginatedArticles.filter((a) => a.externalLink)"
```
Make a variable for this

const externalArticles = computed(() =>
  paginatedArticles.value.filter(a => a.externalLink)
)
const internalArticles = computed(() =>
  paginatedArticles.value.filter(a => !a.externalLink)
)

jo-svg commented on siths-redesign/components/MakerspaceGrid.vue:
```diff
@@ -1,20 +1,16 @@
+      <a
+        v-for="article in paginatedArticles.filter((a) => a.externalLink)"
         :key="article._id"
-        :is="article.externalLink ? 'a' : 'div'"
```
Y delete this

jo-svg replied:
Wouldnt it be better to js run that one part once? Like ion think u need two divs to check if its an internal or external article

jo-svg commented on siths-redesign/components/MakerspaceGrid.vue:
```diff
@@ -105,6 +106,16 @@ const props = defineProps({
+  selectedArticle.value = article
+}
+
+function truncateText(text, maxLength) {
```
I couldve sworn i put this in a folder so u didnt need to rewrite this

katiekhamarkhanova replied:
wait where is it

jo-svg replied:
maybe im bugging

jo-svg commented on siths-redesign/components/MakerspaceGrid.vue:
```diff
@@ -1,20 +1,16 @@
 <template>
   <div class="w-full px-4 lg:w-3/5 xl:w-3/6">
```
Xl:w- 3/6 -> 1/2

jo-svg commented on siths-redesign/components/MakerspaceGrid.vue:
```diff
@@ -1,20 +1,16 @@
+        :href="article.externalLink"
+        target="_blank"
+        rel="noopener noreferrer"
+        class="group cursor-pointer"
```
Cursor-pointer is there for <a> is default in browsers

katiekhamarkhanova replied:
?

katiekhamarkhanova replied:
nvm

jo-svg commented on siths-redesign/components/Pagination.vue:
```diff
@@ -0,0 +1,48 @@
+      @click="currentPage > 1 ? $emit('previous') : null"
+      :class="[
+        'btn inline-flex',
+        currentPage === 1 ? 'cursor-not-allowed opacity-50' : 'cursor-pointer'
```
U can use :disabled for this
If u do use that, inside ur class u can put disabled:opacity-50 disabled:cursor-not-allowed
Also u dont need cursor-pointer bc button alr has cursor-pointer as default

jo-svg commented on siths-redesign/components/Pagination.vue:
```diff
@@ -0,0 +1,48 @@
+    <div class="flex gap-2">
+      <button
+        v-for="page in totalPages"
+        :key="page"
```
U can also add :disabled= "currentpage === page" so u dknt emit page that is alr active

bridgetf7 commented:
Girl get some sleep it’s break

jo-svg commented:
LMFAOO igu imma knock out nowww

kennething commented on siths-redesign/components/Pagination.vue:
```diff
@@ -0,0 +1,48 @@
+  <div v-if="totalPages > 1" class="mt-8 flex w-full items-center justify-center gap-4">
+    <button
+      @click="currentPage > 1 ? $emit('previous') : null"
+      :class="[
```
> \[!CAUTION\]
> vue will keep wasting resources to rerender this string every time `variable` gets changed
> this is what u need to do in react
> **NEVER** do something like react
> if ur doing something like react ur doing it **WRONG**
> react bad

if it doesnt need to change it doesnt need to get v-binded

```rs
class="non changing classes"
:class="boolean ? 'wtv' : 'wtv'"
```

jo-svg replied:
i was wondering why the days since kenneth reseted 💀

kennething commented on siths-redesign/pages/cte/makerspace.vue:
```diff
@@ -1,6 +1,7 @@
 <template>
     <h1 class="mb-4 text-center text-2xl md:text-3xl lg:text-4xl">Career and Technical Education</h1>
     <div class="mb-6 flex flex-col items-start justify-center px-4 md:flex-row">
+        <SubpageMenu :pages="cteLinks" :active="'About'" class="block mb-4 lg:absolute left-8 top-40" />
```
u dont need to v-bind for strings
`active="About"`

katiekhamarkhanova committed:
Merge branch 'dev' into 127-fix-makerspace-components-styling

katiekhamarkhanova committed:
s

joe-238 committed:
im a bum!!
